Core/Src/main.c

```diff
@@ -9,46 +9,57 @@
  * 
  */
 
-#include <string.h>
 #include <stdio.h>
-
-#include "stm32f0xx.h"
-#include "stm32f051_rcc.h"
-#include "stm32f051_gpio.h"
-#include "stm32f051_uart.h"
-#include "stm32f051_i2c.h"
+#include <stdlib.h>
+#include <string.h>
 
 #include "main.h"
 #include "pins.h"
 
 #include "ds3231.h"
+#include "uart_ring.h"
+
+struct
+{
+  char strOut[100];
+  uint8_t count;
+  char bufIn[100];
```

Comment:
Please, add a description for this magic variable `100` or add a definition/enumeration/constant that will hold this number